app/src/main/java/com/internship/move/feature/authentication/ResetPasswordFragment.kt

```diff
@@ -0,0 +1,48 @@
+package com.internship.move.feature.authentication
+
+import android.os.Bundle
+import android.text.Editable
+import android.view.View
+import androidx.core.widget.doOnTextChanged
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import com.internship.move.R
+import com.internship.move.databinding.FragmentResetPasswordBinding
+import com.zhuinden.fragmentviewbindingdelegatekt.viewBinding
+
+class ResetPasswordFragment : Fragment(R.layout.fragment_reset_password) {
+
+    private val binding by viewBinding(FragmentResetPasswordBinding::bind)
+    private val args by navArgs<ResetPasswordFragmentArgs>()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.tokenTV.text = args.token
+        initListeners()
+    }
+
+    private fun initListeners() {
+        binding.newPassTIET.doOnTextChanged { text, _, _, _ ->
+            binding.resetPassBtn.isEnabled = (text?.isNotEmpty() ?: true) and
```

Comment:
This bit of code is repeated twice, you should extracted to a separate method
Also, instead of using `text?.isNotEmpty() ?: true`, you could use `text.isNullOrEmpty()` instead. This is a call on a null value that returns a non-null result, and removes the need for a default `?:` value. You'll have to tweak the check a bit tho